openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+
+            id = self.domains[i].id
+            if not self.by_nuclide:
+                # Build & add metadata to XSdata object
+                # (Use i here because k in nuclides will add chars to this)
+                if domain_names is None:
+                    name = 'set' + str(i + 1)
+                else:
+                    name = domain_names[i]
+                name += '.' + xs_ids[i]
+                xsdata = openmc.XSdata(name, self.energy_groups)
+                xsdata.order = order
+
+                mat_names[id] = name
+
+                # Now get xs data itself
+                if 'transport' in self.mgxs_types:
+                    if self.correction == 'P0':
+                        xsdata.set_total(self.all_mgxs[id]['transport'],
+                                         xs_type=xs_type, subdomains=(id,))
+                    else:
+                        msg = "The use of a transport cross section " + \
+                              "requires the correction attribute to be" + \
+                              "set to 'P0' to produce valid cross " + \
+                              "section libraries"
+                        raise ValueError(msg)
+                elif 'total' in self.mgxs_types:
+                    xsdata.set_total(self.all_mgxs[id]['total'],
+                                     xs_type=xs_type, subdomains=(id,))
+                if 'absorption' in self.mgxs_types:
+                    xsdata.set_absorption(self.all_mgxs[id]['absorption'],
+                                          xs_type=xs_type,
+                                          subdomains=(id,))
+                if 'fission' in self.mgxs_types:
+                    xsdata.set_fission(self.all_mgxs[id]['fission'],
+                                       xs_type=xs_type, subdomains=(id,))
+                if 'kappa-fission' in self.mgxs_types:
+                    xsdata.set_k_fission(self.all_mgxs[id]['kappa-fission'],
+                                         xs_type=xs_type, subdomains=(id,))
+                if 'chi' in self.mgxs_types:
+                    xsdata.set_chi(self.all_mgxs[id]['chi'],
+                                   xs_type=xs_type, subdomains=(id,))
+                if 'nu-fission' in self.mgxs_types:
+                    xsdata.set_nu_fission(self.all_mgxs[id]['nu-fission'],
+                                          xs_type=xs_type,
+                                          subdomains=(id,))
+                # multiplicity requires scatter and nu-scatter
+                if ((('scatter matrix' in self.mgxs_types) and
+                     ('nu-scatter matrix' in self.mgxs_types))):
+                    xsdata.set_multiplicity(
+                        self.all_mgxs[id]['nu-scatter matrix'],
+                        self.all_mgxs[id]['scatter matrix'],
+                        xs_type=xs_type, subdomains=(id,))
+                    xsdata.multiplicity = np.nan_to_num(xsdata.multiplicity)
```

Comment:
Should the `nan_to_num(...)` be buried within the `XSdata.set_multiplicity(...)` method?